File: kant_knowledge_base/kant_knowledge_base/knowledge_base_node_main.py
#!/usr/bin/env python3

import rclpy
from kant_knowledge_base.knowledge_base import KnowledgeBaseNode


def main(args=None):
    rclpy.init(args=args)

    node = KnowledgeBaseNode()

    node.join_spin()

    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: kant_knowledge_base/src/knowledge_base_node_main.cpp

#include <memory>

#include "kant_knowledge_base/knowledge_base/knowledge_base_node.hpp"
#include "simple_node/node.hpp"

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<
      kant::knowledge_base::knowledge_base::KnowledgeBaseNode>();
  rclcpp::spin(node->get_node_base_interface());
  rclcpp::shutdown();
  return 0;
}